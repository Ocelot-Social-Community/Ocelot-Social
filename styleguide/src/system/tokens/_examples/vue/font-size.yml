#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
