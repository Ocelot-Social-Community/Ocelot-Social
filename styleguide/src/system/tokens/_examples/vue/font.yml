#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
