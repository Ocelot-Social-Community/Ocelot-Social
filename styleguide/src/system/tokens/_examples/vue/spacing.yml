#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
