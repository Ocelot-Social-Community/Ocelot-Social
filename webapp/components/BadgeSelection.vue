<template>
  <div class="badge-selection">
    <button
      v-for="(badge, index) in badges"
      :key="badge.id"
      class="badge-selection-item"
      @click="handleBadgeClick(badge, index)"
    >
      <div class="badge-icon">
        <img :src="badge.icon | proxyApiUrl" :alt="badge.id" />
      </div>
      <div class="badge-info">
        <div class="badge-description">{{ badge.description }}</div>
      </div>
    </button>
  </div>
</template>

<script>
export default {
  name: 'BadgeSelection',
  props: {
    badges: {
      type: Array,
      default: () => [],
    },
  },
  data() {
    return {
      selectedIndex: null,
    }
  },
  methods: {
    handleBadgeClick(badge, index) {
      if (this.selectedIndex === index) {
        this.selectedIndex = null
        this.$emit('badge-selected', null)
        return
      }

      this.selectedIndex = index
      this.$emit('badge-selected', badge)
    },
    resetSelection() {
      this.selectedIndex = null
    },
  },
}
</script>

<style lang="scss" scoped>
.badge-selection {
  width: 100%;
  max-width: 600px;
  margin: 0 auto;

  .badge-selection-item {
    display: flex;
    align-items: center;
    padding: 12px 16px;
    margin-bottom: 8px;
    border-radius: 8px;
    background-color: #f5f5f5;
    transition: all 0.2s ease;
    width: 100%;
    text-align: left;
    cursor: pointer;

    &:hover {
      background-color: #e0e0e0;
    }

    .badge-icon {
      flex-shrink: 0;
      width: 40px;
      height: 40px;
      margin-right: 16px;

      img {
        width: 100%;
        height: 100%;
        object-fit: contain;
      }
    }

    .badge-info {
      flex-grow: 1;

      .badge-title {
        font-weight: bold;
        font-size: 16px;
        margin-bottom: 4px;
      }

      .badge-description {
        font-size: 14px;
        color: #666;
      }
    }
  }
}
</style>
