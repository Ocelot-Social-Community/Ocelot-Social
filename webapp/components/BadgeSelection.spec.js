import { render, screen, fireEvent } from '@testing-library/vue'
import BadgeSelection from './BadgeSelection.vue'

const localVue = global.localVue

describe('Badges.vue', () => {
  const Wrapper = (propsData) => {
    return render(BadgeSelection, {
      propsData,
      localVue,
    })
  }

  describe('without badges', () => {
    it('renders', () => {
      const wrapper = Wrapper({ badges: [] })
      expect(wrapper.container).toMatchSnapshot()
    })
  })

  describe('with badges', () => {
    const badges = [
      {
        id: '1',
        icon: '/path/to/some/icon',
        isDefault: false,
        description: 'Some description',
      },
      {
        id: '2',
        icon: '/path/to/another/icon',
        isDefault: true,
        description: 'Another description',
      },
      {
        id: '3',
        icon: '/path/to/third/icon',
        isDefault: false,
        description: 'Third description',
      },
    ]

    let wrapper

    beforeEach(() => {
      wrapper = Wrapper({ badges })
    })

    it('renders', () => {
      expect(wrapper.container).toMatchSnapshot()
    })

    describe('clicking on a badge', () => {
      beforeEach(async () => {
        const badge = screen.getByText(badges[1].description)
        await fireEvent.click(badge)
      })

      it('emits badge-selected with badge', async () => {
        expect(wrapper.emitted()['badge-selected']).toEqual([[badges[1]]])
      })
    })

    describe('clicking twice on a badge', () => {
      beforeEach(async () => {
        const badge = screen.getByText(badges[1].description)
        await fireEvent.click(badge)
        await fireEvent.click(badge)
      })

      it('emits badge-selected with null', async () => {
        expect(wrapper.emitted()['badge-selected']).toEqual([[badges[1]], [null]])
      })
    })
  })
})
