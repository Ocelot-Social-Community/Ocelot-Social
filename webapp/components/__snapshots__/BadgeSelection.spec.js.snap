// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badges.vue with badges renders 1`] = `
<div>
  <div
    class="badge-selection"
  >
    <button
      class="badge-selection-item"
    >
      <div
        class="badge-icon"
      >
        <img
          alt="1"
          src="/api/path/to/some/icon"
        />
      </div>
       
      <div
        class="badge-info"
      >
        <div
          class="badge-description"
        >
          Some description
        </div>
      </div>
    </button>
    <button
      class="badge-selection-item"
    >
      <div
        class="badge-icon"
      >
        <img
          alt="2"
          src="/api/path/to/another/icon"
        />
      </div>
       
      <div
        class="badge-info"
      >
        <div
          class="badge-description"
        >
          Another description
        </div>
      </div>
    </button>
    <button
      class="badge-selection-item"
    >
      <div
        class="badge-icon"
      >
        <img
          alt="3"
          src="/api/path/to/third/icon"
        />
      </div>
       
      <div
        class="badge-info"
      >
        <div
          class="badge-description"
        >
          Third description
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`Badges.vue without badges renders 1`] = `
<div>
  <div
    class="badge-selection"
  />
</div>
`;
